### DISCUSSION TITLE
Your device ran into a problem... (Only during UEFI iPXE)

### DISCUSSION THREAD

**Discussion by SuneThomsenDK** [ipxe/ipxe#325] [Q&A]
Hi,

I'm testing a boot.wim and iPXE (UEFI) but keep getting this ERROR after awhile and I can´t figure out why this is happening?

![PXE_Boot](https://user-images.githubusercontent.com/47953133/115014457-a1bfc900-9eb2-11eb-9101-a01b5bd96aaf.JPG)

1. It´s only happening during UEFI sesion (ipxe.efi)
2. iPXE BIOS legacy (undionly.kpxe) does work without any errors.
3. WDS PXE does also work without any errors.
4. ISO format mounted to my Hyper-V does also work without any errors.

What kind of information do you need from me in order to try and help me out ?

Thanks in advance :)

**Comment by NiKiZe** [comment]
This is during boot of winpe by using wimboot? Post your iPXE script. Make sure you have tested latest versions of iPXE and wimboot, and finally try with a unmodified boot.wim from Windows base iso.

**Reply by SuneThomsenDK** [reply]
Hi, 

Not during boot, WinPE actually boot fine, but after awhile (4-5 minutes) in WinPE I'll get that ERROR, but only in UEFI (Non-Secure Boot mode). I'm using latest wimboot (version 2.7.2) and latest ipxe.efi / undionly.kpxe files.

Just to make sure, can you link the latest wimboot, ipxe.efi and undionly.kpxe files? Also I know that wimboot is signed and that ipxe.efi is not, so thats why I run UEFI in Non-Secure Boot mode during my tests.

iPXE Script:
------------
#!ipxe
kernel http://ipxe/wim/wimboot gui
initrd -n BCD http://ipxe/wim/media/boot/bcd BCD
initrd -n boot.sdi http://ipxe/wim/media/boot/boot.sdi boot.sdi
initrd -n boot.wim http://ipxe/wim/media/sources/boot.wim boot.wim
boot

I'll try with a unmodified boot.wim from Windows base ISO and see what happens.

Thanks :)

**Reply by NiKiZe** [reply]
boot.ipxe.org and the wimboot repo has latest prebuilt files.

However with that explanation I would start to think that this is a watchdog timer sending a NMI event. Maybe iPXE enables that timer, or maybe your scenario for some reason disables the reset in winpe. On what machine is this?

Have you looked for any other information about NMI_HARDWARE_FAILURE? (NMI is a nice but niche feature) if it is what I think it is then some triggers NMI exception (externally) but why is what you need to figure out.

**Reply by SuneThomsenDK** [reply]
Bear with me, I'm new to this, so I'll have to read up on things like watchdog timer and NMI events :)
Keep in mind that the same boot.wim is working perfect during BIOS legacy iPXE boot (using the undionly.kpxe file) and normal WDS PXE boot.

It´s an Hyper-V build on-top of a Lenovo X1 Extreme Gen 3.

I've just updated to the latest ipxe.efi and undionly.kpxe, the wimboot file were the latest. Then I will look into any other information about NMI_HARDWARE_FAILURE.

Thanks again for your patience and help 👍

**Reply by NiKiZe** [reply]
What I'm trying to say is that iPXE is not causing the NMI event itself, something else is, the question is why, and what you can do about it. Quick search found this, which does explain I bit about the basics here. https://docs.microsoft.com/en-us/troubleshoot/windows-client/performance/nmi-hardware-failure-error

**Reply by SuneThomsenDK** [reply]
Awesome 👍 Thanks mate, I've also found some explanation on the topic myself, which made me wonder if it were hardware related, could it be driver issue? I removed the Dell WinPE drivers from the boot.wim and I've successfully booted and completed the task in UEFI mode a few times now on my Hyper-V :D
